### ISSUE TITLE
Building amplify_authenticator for IOS fails

### ISSUE THREAD

**Issue by Lance-Kraakman** [aws-amplify/amplify-flutter#3595]
### Description

Cannot build for IOS with following error.
Works fine on android.

![image](https://github.com/aws-amplify/amplify-flutter/assets/44858574/77067328-3822-4805-9bd5-a1d3cba6f4a4)

I believe everything is setup as per the documentation.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [X] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Screenshots

![image](https://github.com/aws-amplify/amplify-flutter/assets/44858574/508d4f8d-04f4-4fc3-abb3-14f7c7f7613e)

![image](https://github.com/aws-amplify/amplify-flutter/assets/44858574/edd224f0-5bd6-488b-9a4d-9c30e08e884e)

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.3.0

### Amplify Flutter Version

1.3.2

### Deployment Method

Amplify CLI

**Comment by Equartey** [comment]
Hey @Lance-Kraakman, sorry to hear you're facing build issues.

Can you share your amplify config (you can remove any sensitive or app specific info such as API keys or app IDs)?

**Comment by Lance-Kraakman** [comment]
Sure:

amplifyconfiguration
```
const amplifyconfig = '''{
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "valiturustraining": {
                    "endpointType": "GraphQL",
                    "endpoint": "endpoint",
                    "region": "ap-southeast-2",
                    "authorizationType": "AMAZON_COGNITO_USER_POOLS",
                    "apiKey": "apiKey"
                }
            }
        }
    },
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "AppSync": {
                    "Default": {
                        "ApiUrl": "endpoint",
                        "Region": "ap-southeast-2",
                        "AuthMode": "AMAZON_COGNITO_USER_POOLS",
                        "ClientDatabasePrefix": "valiturustraining_AMAZON_COGNITO_USER_POOLS"
                    },
                    "valiturustraining_API_KEY": {
                        "ApiUrl": "endpoint",
                        "Region": "ap-southeast-2",
                        "AuthMode": "API_KEY",
                        "ApiKey": "apiKey",
                        "ClientDatabasePrefix": "valiturustraining_API_KEY"
                    },
                    "valiturustraining_AWS_IAM": {
                        "ApiUrl": "endpoint",
                        "Region": "ap-southeast-2",
                        "AuthMode": "AWS_IAM",
                        "ClientDatabasePrefix": "valiturustraining_AWS_IAM"
                    }
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": "********",
                            "Region": "ap-southeast-2"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "poolID",
                        "AppClientId": "********",
                        "Region": "ap-southeast-2"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH",
                        "mfaConfiguration": "OFF",
                        "mfaTypes": [
                            "SMS"
                        ],
                        "passwordProtectionSettings": {
                            "passwordPolicyMinLength": 8,
                            "passwordPolicyCharacters": [
                                "REQUIRES_LOWERCASE",
                                "REQUIRES_NUMBERS",
                                "REQUIRES_SYMBOLS",
                                "REQUIRES_UPPERCASE"
                            ]
                        },
                        "signupAttributes": [
                            "NAME",
                            "EMAIL",
                            "GENDER",
                            "PHONE_NUMBER",
                            "BIRTHDATE"
                        ],
                        "socialProviders": [],
                        "usernameAttributes": [
                            "EMAIL"
                        ],
                        "verificationMechanisms": [
                            "EMAIL"
                        ]
                    }
                }
            }
        }
    }
}''';
```

backend-json
{
  "api": {
    "valiturustraining": {
      "dependsOn": [
        {
          "attributes": [
            "UserPoolId"
          ],
          "category": "auth",
          "resourceName": "valiturustraining"
        }
      ],
      "output": {
        "authConfig": {
          "additionalAuthenticationProviders": [
            {
              "apiKeyConfig": {
                "apiKeyExpirationDate": "2024-07-21T11:14:28.596Z",
                "apiKeyExpirationDays": 365,
                "description": "graphql api key"
              },
              "authenticationType": "API_KEY"
            },
            {
              "authenticationType": "AWS_IAM"
            }
          ],
          "defaultAuthentication": {
            "authenticationType": "AMAZON_COGNITO_USER_POOLS",
            "userPoolConfig": {
              "userPoolId": "authvaliturustraining"
            }
          }
        }
      },
      "providerPlugin": "awscloudformation",
      "service": "AppSync"
    }
  },
  "auth": {
    "valiturustraining": {
      "customAuth": false,
      "dependsOn": [],
      "frontendAuthConfig": {
        "mfaConfiguration": "OFF",
        "mfaTypes": [
          "SMS"
        ],
        "passwordProtectionSettings": {
          "passwordPolicyCharacters": [
            "REQUIRES_LOWERCASE",
            "REQUIRES_NUMBERS",
            "REQUIRES_SYMBOLS",
            "REQUIRES_UPPERCASE"
          ],
          "passwordPolicyMinLength": 8
        },
        "signupAttributes": [
          "NAME",
          "EMAIL",
          "GENDER",
          "PHONE_NUMBER",
          "BIRTHDATE"
        ],
        "socialProviders": [],
        "usernameAttributes": [
          "EMAIL"
        ],
        "verificationMechanisms": [
          "EMAIL"
        ]
      },
      "providerPlugin": "awscloudformation",
      "service": "Cognito"
    }
  },
  "function": {
    "generateWorkout": {
      "build": true,
      "dependsOn": [
        {
          "attributes": [
            "GraphQLAPIIdOutput",
            "GraphQLAPIEndpointOutput",
            "GraphQLAPIKeyOutput"
          ],
          "category": "api",
          "resourceName": "valiturustraining"
        },
        {
          "attributes": [
            "UserPoolId"
          ],
          "category": "auth",
          "resourceName": "valiturustraining"
        },
        {
          "attributes": [
            "Arn"
          ],
          "category": "function",
          "resourceName": "valiturustrainingvaliturusLayer"
        }
      ],
      "providerPlugin": "awscloudformation",
      "service": "Lambda"
    },
    "valiturustrainingvaliturusLayer": {
      "build": true,
      "providerPlugin": "awscloudformation",
      "service": "LambdaLayer"
    }
  },
  "parameters": {
    "AMPLIFY_function_generateWorkout_deploymentBucketName": {
      "usedBy": [
        {
          "category": "function",
          "resourceName": "generateWorkout"
        }
      ]
    },
    "AMPLIFY_function_generateWorkout_s3Key": {
      "usedBy": [
        {
          "category": "function",
          "resourceName": "generateWorkout"
        }
      ]
    },
    "AMPLIFY_function_valiturustrainingvaliturusLayer_deploymentBucketName": {
      "usedBy": [
        {
          "category": "function",
          "resourceName": "valiturustrainingvaliturusLayer"
        }
      ]
    },
    "AMPLIFY_function_valiturustrainingvaliturusLayer_s3Key": {
      "usedBy": [
        {
          "category": "function",
          "resourceName": "valiturustrainingvaliturusLayer"
        }
      ]
    }
  }
}

**Comment by dnys1** [comment]
Hi @Lance-Kraakman, it seems you may be using the `master` branch of Flutter. The change noted in the error message was only recently introduced to the Flutter source code.

We will investigate this as it will affect users in the future, but switching to a stable or beta version of Flutter should resolve it for you in the short term.

**Comment by Equartey** [comment]
Hey @Lance-Kraakman, a fix for this was included in our latest release [1.4.0](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.4.0). Thanks again for identifying it!

**Comment by Lance-Kraakman** [comment]
> Hey @Lance-Kraakman, a fix for this was included in our latest release [1.4.0](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.4.0). Thanks again for identifying it!

No stress!, thanks for the speedy support!
